client/state/push-notifications/selectors.js

```diff
@@ -0,0 +1,73 @@
+
+export const isApiReady = ( state ) => !! state.pushNotifications.settings.apiReady;
+export const isAuthorized = ( state ) => !! state.pushNotifications.settings.authorized;
+export const isBlocked = ( state ) => !! state.pushNotifications.settings.blocked;
+export const isEnabled = ( state ) => !! state.pushNotifications.settings.enabled;
+export const isNoticeDismissed = ( state ) => !! state.pushNotifications.settings.dismissedNotice;
+export const isShowingUnblockInstructions = ( state ) => !! state.pushNotifications.settings.showingUnblockInstructions;
+export const getSavedSubscription = ( state ) => state.pushNotifications.settings.subscription;
+export const getSavedWPCOMSubscription = ( state ) => state.pushNotifications.settings.wpcomSubscription;
+export const getLastUpdated = ( state ) => state.pushNotifications.settings.lastUpdated;
+
+export function isPushNotificationsSupported() {
+	return (
+		isServiceWorkerSupported() &&
+		'showNotification' in window.ServiceWorkerRegistration.prototype &&
+		'PushManager' in window
+	);
+}
+
+export function isServiceWorkerSupported() {
+	return (
+		'serviceWorker' in window.navigator &&
+		'ServiceWorkerRegistration' in window
+	);
+}
+
+export function isPushNotificationsDenied() {
```

Comment:
Look into using this function to guard against unexpected behavior when dispatching `block()`